plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java

```diff
@@ -105,6 +112,9 @@ public static Type createTypeFromCoercer(TypeManager typeManager, HiveType fromH
             }
             return Optional.empty();
         }
+        if (fromType instanceof VarcharType fromVarcharType && toType instanceof DateType toDateType) {
```

Comment:
Aren't you missing also  in https://github.com/trinodb/trino/blob/79ea8354c44bb5c04638226fa3e3dd4b5ac054f4/plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeTranslator.java#L39


```
        if (isVarcharType(fromOrcType) && toTrinoType instanceof DateType dateType) {
            return Optional.of(new DateCoercer.VarcharToDateCoercer(createUnboundedVarcharType(), dateType));
        }
```

?

Reply:
I was not planning to do it for unpartitioned tables. For partitioned tables this is not required.

Reply:
@Praveen2112  why do we have the code duplication for coercion in the `OrcTypeTranslator` ?